lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -2535,33 +2553,33 @@ prop_balanceTransactionBalanced
     -> ShowBuildable PartialTx
     -> StdGenSeed
     -> Property
-prop_balanceTransactionBalanced (Wallet' utxo wal pending) (ShowBuildable partialTx) seed
-    = withMaxSuccess 200 $ do
+prop_balanceTransactionBalanced wallet (ShowBuildable partialTx') seed
+    = withMaxSuccess 1000 $ do
         let combinedUTxO = mconcat
                 [ resolvedInputsUTxO Cardano.ShelleyBasedEraAlonzo partialTx
-                , toCardanoUTxO Cardano.ShelleyBasedEraAlonzo $ view #utxo wal
+                , toCardanoUTxO Cardano.ShelleyBasedEraAlonzo walletUTxO
                 ]
         let originalBalance = txBalance (sealedTx partialTx) combinedUTxO
         let res = balanceTransaction'
-                (utxo, wal, pending)
+                wallet
                 seed
                 partialTx
         case res of
             Right sealedTx -> counterexample ("\nResult: " <> pretty sealedTx) $ do
                 label "success"
-                    $ classify (originalBalance == Cardano.Lovelace 0)
+                    $ classify (originalBalance == mempty)
                         "already balanced"
                     $ classify (txFee sealedTx > Cardano.Lovelace 1_000_000)
                         "fee above 1 ada"
-                    $ classify (txFee sealedTx > Cardano.Lovelace 1_000_000)
-                        "fee above 1 ada"
                     $ classify (hasCollateral sealedTx)
                         "balanced tx has collateral"
-                    (txBalance sealedTx combinedUTxO === 0)
-                    .&&. (abs (txFee sealedTx) .<= 4_000_000)
-                    -- Fee limit chosen at a hunch for the sake of sanity. As
-                    -- long as the property uses mainnet PParams, this is
-                    -- useful.
+                    $ conjoin
+                        [ txBalance sealedTx combinedUTxO === mempty
+                        , (prop_expectFeeExcessSmallerThan
+                            (Cardano.Lovelace 1_200_000) sealedTx)
```

Comment:
Perhaps it would be worth adding a comment here to describe the reasoning behind the 1.2 ada value?

Reply:
2c533e2cd312ffc9e505a9be79ee5a7cdd55a49e

Reply:
2c533e2cd312ffc9e505a9be79ee5a7cdd55a49e